Makefile

```diff
@@ -0,0 +1,34 @@
+# Makefile
+
+## Additionaly arguments passed to cargo.
+EXTRA_CARGO_ARGS ?=
+## Whether to disable nightly-only feature. [true/false]
+WITH_STABLE_TOOLCHAIN ?=
+
+.PHONY: format clippy test
+
+all: format clippy test
+
+## Format code in-place using rustfmt.
+format:
+	cargo fmt --all
+
+## Run clippy.
+ifeq ($(WITH_STABLE_TOOLCHAIN), true)
+clippy:
+	cargo clippy --all --features all_stable --all-targets -- -D clippy::all
+else
+clippy:
+	cargo clippy --all --all-features --all-targets -- -D clippy::all
```

Comment:
Here, do we not need `+nightly` trait any more when `${WITH_STABLE_TOOLCHAIN} == false` ?

I've re-test the makefile on Centos:
`Linux version 3.10.0-862.14.4.el7.x86_64 (mockbuild@kbuilder.bsys.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC) ) #1 SMP Wed Sep 26 15:12:11 UTC 2018`

And it returns the following failure: 
```Rust
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> src/lib.rs:16:34
   |
16 | #![cfg_attr(feature = "nightly", feature(test))]
   |                                  ^^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> src/lib.rs:17:31
```

And if I add the `+nightly` trait, both `make test` and `make clippy` worked as expectation without errs, by setting `${WITH_STABLE_TOOLCHAIN} == false` .

Reply:
The idea is to assume user has set the correct default toolchain. But force setting sounds okay too, I'll add it later.